PhilanthropyDataCommons/front-end
# Add BulkUpload history components

This PR adds the BulkUpload history components and wires up the history to the API.

This is an incomplete PR but I'm hopping on a plane. Hopefully this establishes enough patterns and styling for you to continue from here! Still clearly TODO:

- [x] Demonstrate this in a panel
- [x] Humanize the byte size (1KB, 2MB, 1.1GB, etc.)
- [x] Humanize the time format
   - [x] Wrap the time in a `<time>` element with `datetime` attribute
- [x] Adapt to the real BulkUpload object attributes

**Update, January 19:**

Since writing the above, here's what's happened: Jason took over this branch, loaded BulkUpload history from the API, and wired up the components. (He also added file uploading on top of it, but that will be in a separate PR.) I collected that work into commits and added the humanized times and file sizes.

**Testing:**
- Check out the Storybook components
- Locally, create some BulkUpload history in your database (apologies, uploading itself will be in a forthcoming PR)
- Make sure the sidebar displays them correctly!

Note that the uploader itself is not included in this PR, and the styling/behavior of the components does not yet include the changes suggested in #477.

Closes #413
Closes #414
Closes #492
